third_party/move/move-prover/move-docgen/src/docgen.rs

```diff
@@ -719,6 +719,133 @@ impl<'env> Docgen<'env> {
+
+        for row_block in row_blocks {
+            if !row_block.trim().is_empty() {
+                self.gen_spec_rows(row_block, column_names.clone());
+            }
+        }
+        self.doc_text("</table>\n");
+    }
+
+    fn gen_spec_rows(&self, row_block: &str, column_names: Vec<&str>) {
+        let lines = row_block.lines().collect::<Vec<_>>();
+        let mut row_data = vec![String::new(); column_names.len()];
+        let mut current_key: Option<usize> = None;
+
+        for line in lines {
+            let trimmed_line = line.trim();
+            if trimmed_line.is_empty() {
+                continue;
+            }
+
+            let parts = trimmed_line.splitn(2, ':').collect::<Vec<_>>();
+            if parts.len() == 2 {
+                let key = parts[0].trim();
+
+                if let Some(index) = column_names.iter().position(|&name| name == key) {
+                    let value = self.convert_to_anchor(parts[1].trim());
+                    row_data[index] = value;
+                    current_key = Some(index);
+                }
+            } else if let Some(key_index) = current_key {
+                row_data[key_index].push(' ');
+                row_data[key_index].push_str(&self.convert_to_anchor(trimmed_line));
+            }
+        }
+
+        self.doc_text(&format!(
+            "<tr>\n{}\n</tr>\n",
+            row_data
+                .iter()
+                .map(|data| format!("<td>{}</td>", data))
+                .collect::<Vec<_>>()
+                .join("\n")
+        ));
+    }
+
+    fn gen_spec_tag(&self, tag: &str) {
+        let (spec_tag, module_link, suffix) = if tag.contains("::") {
+            let parts = tag.split("::").collect::<Vec<_>>();
+            let module_name = *parts.first().unwrap_or(&"");
+            let spec_tag = *parts.get(1).unwrap_or(&"");
+            let label_link = self
+                .resolve_to_label(module_name, false)
+                .unwrap_or(String::new());
+            let module_link = label_link.split('#').next().unwrap_or("").to_string();
+            let suffix = format!(" of the <a href={}>{}</a> module", module_link, module_name);
+            (spec_tag, module_link, suffix)
+        } else {
+            (tag, String::new(), String::new())
+        };
+
+        let spec_number = spec_tag
+            .split('-')
+            .nth(3)
+            .unwrap_or_default()
+            .split('.')
+            .next()
+            .unwrap_or_default();
+        self.doc_text_general(false, &format!("// This enforces <a id={} href=\"{}#high-level-spec\">high-level specification {}</a>{}:", tag, module_link, spec_number, suffix));
+    }
+
+    fn convert_to_anchor(&self, input: &str) -> String {
+        let mut result = String::new();
+        let mut last_index = 0;
+
+        let mut i = 0;
+        while i < input.len() {
+            if let Some(start) = input[i..].find('(') {
```

Comment:
I am wondering we could use regular expression matching here? This is a reference: https://github.com/aptos-labs/aptos-core/blob/eecbcfc1c627048418a0e62e59aa8d69803cc6cd/third_party/move/move-prover/boogie-backend/src/boogie_wrapper.rs#L101

Reply:
I agree that this would be better, but since we need this to move forward with actually using this, the current technique is also OK with me.

Reply:
Done. Thanks!